### PR TITLE
feat: add sed as tool dependency and support 1.29.4

### PR DESCRIPTION
**Pull request by alessiodionisi**
- Added `sed` as dependency
- Bumped go mod fury-distribution dependency to 1.29.4

Related to https://github.com/sighupio/fury-distribution/pull/257, https://github.com/sighupio/fury-distribution/pull/261, https://github.com/sighupio/fury-distribution/pull/259.

⚠️  Merge after fury-distribution releases.

### CODE REVIEW

**Review by ralgozino — COMMENTED** [review]
The PR itself lgtm, I'm "requesting changes" until we address the open points in the linked PR.

Just to confirm, your idea is to backport the feature to the last releases with this PR and **not** to do a patch release of the distro?

**Comment by ralgozino** [comment]
To answer myself, I just tried to create a cluster using KFD v1.29.3 and a build of furyctl from this branch and I get an error on the schema:

```bash
$ ~/src/github.com/sighupio/furyctl/furyctl apply cluster --phase kubernetes --config customRegistry.yaml
DEBU logging to: <REDACTED>/.furyctl/furyctl.1725959515-51887.log
DEBU Getting Home Directory Path...
INFO Downloading distribution...
INFO Compatibility patches applied for v1.29.3
INFO Validating configuration file...
ERRO error while validating configuration file: error while validating against schema: jsonschema: '/spec/kubernetes/advanced' does not validate with file:///var/folders/lb/zscvwt_s6fx6qdrfv4g51dsc0000gn/T/furyctl-2425192853/data/schemas/public/onpremises-kfd-v1alpha2.json#/properties/spec/$ref/properties/kubernetes/$ref/properties/advanced/$ref/additionalProperties: additionalProperties 'customRegistry' not allowed
```

**Review by ralgozino — CHANGES_REQUESTED** [review]
a per previous comments

**Comment by alessiodionisi** [comment]
@ralgozino backports removed

**Review by ralgozino — DISMISSED** [review]
changes related to the mirror feature lgtm.

BUT these changes are not enough for releasing, for example bumping the 1.28.x and 1.27.y max version is missing

**Comment by alessiodionisi** [comment]
I think the PR is ready to release v0.29.7

File: docs/COMPATIBILITY_MATRIX.md [hunk 1]
```diff
@@ -7,30 +7,32 @@ Note: Always use the latest `furyctl` version, we make sure that is compatible w
...
+| 0.29.6        |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
+| 0.29.5        | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
```
Comment (ralgozino) [inline docs/COMPATIBILITY_MATRIX.md:14]:
```suggestion
| 0.29.5        |                    | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
```

File: docs/COMPATIBILITY_MATRIX.md [hunk 2]
```diff
@@ -7,30 +7,32 @@ Note: Always use the latest `furyctl` version, we make sure that is compatible w
...
+| 0.29.2        | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
+| 0.29.1        | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
```
Comment (ralgozino) [inline docs/COMPATIBILITY_MATRIX.md:18]:
```suggestion
| 0.29.2        |                    | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
| 0.29.1        |                    | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          | :warning:          |
```

File: go.mod [hunk 3]
```diff
@@ -21,7 +21,7 @@ require (
 	github.com/r3labs/diff/v3 v3.0.1
 	github.com/santhosh-tekuri/jsonschema/v5 v5.3.1
 	github.com/shirou/gopsutil/v3 v3.24.1
-	github.com/sighupio/fury-distribution v1.29.3
+	github.com/sighupio/fury-distribution v1.29.4-rc.3
```
Comment (ralgozino) [inline go.mod:24]: change this with the actual release once it's available

**Review by ralgozino — CHANGES_REQUESTED** [review]

**Review by ralgozino — CHANGES_REQUESTED** [review]
The upgrade paths need to be added in `configs/upgrades`

**Comment by alessiodionisi** [comment]
Added upgrade paths:
- 1.27.8 -> 1.27.9
- 1.28.3 -> 1.28.4
- 1.29.3 -> 1.29.4

minor upgrade paths:
- 1.27.9 -> 1.28.4
- 1.28.4 -> 1.29.4

**Pull request merged**